src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeMemory.Unix.cs

```diff
@@ -0,0 +1,149 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Numerics;
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>This class contains methods that are mainly used to manage native memory.</summary>
+    public static unsafe class NativeMemory
+    {
+        /// <summary>Allocates an aligned block of memory of the specified size and alignment, in bytes.</summary>
+        /// <param name="byteCount">The size, in bytes, of the block to allocate. This should be a multiple of <paramref name="alignment" /> for portability.</param>
+        /// <param name="alignment">The alignment, in bytes, of the block to allocate. This must be a power of <c>2</c> greater than or equal to <c>sizeof(void*)</c> for portability.</param>
+        /// <returns>A pointer to the allocated aligned block of memory.</returns>
+        /// <exception cref="OutOfMemoryException">Allocating <paramref name="byteCount" /> of memory failed.</exception>
+        /// <remarks>
+        ///     <para>This method allows <paramref name="byteCount" /> to be <c>0</c> and will return a valid pointer that should not be dereferenced and that should be passed to free to avoid memory leaks.</para>
+        ///     <para>This method is a thin wrapper over the C <c>aligned_alloc</c> API or a platform dependent aligned allocation API such as <c>_aligned_malloc</c> on Win32.</para>
+        ///     <para>This method is not compatible with <see cref="Free" /> or <see cref="Realloc" />, instead <see cref="AlignedFree" /> should be called.</para>
+        /// </remarks>
+        [CLSCompliant(false)]
+        public static void* AlignedAlloc(nuint byteCount, nuint alignment)
+        {
+            void* result = null;
+
+            // The C standard doesn't define what a valid alignment is, however POSIX requires a power of 2 >= sizeof(void*)
+            if (BitOperations.IsPow2(alignment) && (alignment >= (uint)sizeof(void*)))
```

Comment:
This only validates the alignment is a power of 2 as that is relatively cheap. Also validating that `byteCount % alignment == 0` would be fairly expensive.

Reply:
>  Also validating that byteCount % alignment == 0 would be fairly expensive.

You do not need to use `%` once you know that the alignment is power of 2.

Reply:
I'll also add in a check for `byteCount` since that can just be `(byteCount & (alignment - 1)) == 0`